src/Lean/Linter/Deprecated.lean

```diff
@@ -42,3 +42,18 @@ def checkDeprecated [Monad m] [MonadEnv m] [MonadLog m] [AddMessageContext m] [M
     | none => pure ()
     | some none => logWarning <| .tagged ``deprecatedAttr m!"`{declName}` has been deprecated"
     | some (some newName) => logWarning <| .tagged ``deprecatedAttr m!"`{declName}` has been deprecated, use `{newName}` instead"
+
+-- The following deprecations are have their replacements defined in earlier files,
+-- and as `deprecated` can not be applied to imported declarations,
+-- we have moved the old versions here.
+
+/-- Deprecated synyonym for `congr_arg`. -/
+@[deprecated congr_arg]
+theorem congrArg {α : Sort u} {β : Sort v} {a₁ a₂ : α} (f : α → β) (h : Eq a₁ a₂) : Eq (f a₁) (f a₂) :=
```

Comment:
Why can't this be in Init again? The `deprecated` attribute should be built-in, in which case it should be possible to write this even in the prelude.

Reply:
When I try this in `Init` I get the error message `invalid `[deprecated]` attribute`. (Same if I leave out the identifier.)

The relevant code is:
```
     match stx with
     | `(attr| deprecated $[$id?]?) =>
       let some id := id? | return none
       let declNameNew ← Elab.resolveGlobalConstNoOverloadWithInfo id
       return some declNameNew
     | _  => throwError "invalid `[deprecated]` attribute"
```
Note sure why that `match` is misbehaving in `Init.Prelude`.

Reply:
It's because the `deprecated` attribute itself is defined in `Init/Notation`, it's not a builtin. (Maybe it should be.) But it means that you should be able to place this declaration anywhere after the `syntax` declaration there.

Reply:
Done.